code/modules/mob/mob_movement.dm

```diff
@@ -123,7 +120,11 @@
 	//We are now going to move
 	var/add_delay = mob.cached_multiplicative_slowdown
 	mob.set_glide_size(DELAY_TO_GLIDE_SIZE(add_delay * ( (NSCOMPONENT(direct) && EWCOMPONENT(direct)) ? SQRT_2 : 1 ) )) // set it now in case of pulled objects
-	if(old_move_delay + (add_delay*MOVEMENT_DELAY_BUFFER_DELTA) + MOVEMENT_DELAY_BUFFER > world.time)
+	//If the move was recent, count using old_move_delay
+	//We want fractional behavior and all
+	if(old_move_delay + world.tick_lag > world.time)
+		//Yes this makes smooth movement stutter if add_delay is too fractional
```

Comment:
What does "too fractional" mean?

Reply:
if add_delay is like 0.75, and tick_lag is 0.5, then we step at these times
0ds, 1ds, 1.5ds, 2.5ds 3ds
the delay in time here is not consistent, smooth movement will stutter, so there'd be stops in between moves since the glide delay is lower then the functional move delay
See https://github.com/tgstation/tgstation/pull/28616

The only real way to fix this and preserve non perfect multiple of 0.5 move delays is to use pixel movement
Holding out for a hero, threads will save us I'm sure

Reply:
How odd